### ISSUE TITLE
stackoverflow when using autocompletion

### ISSUE THREAD

**Issue by zhiburt** [rust-analyzer/rust-analyzer#6108]
```
thread '<unknown>' has overflowed its stack
fatal runtime error: stack overflow
```

### Issue

![2020-10-01-14-54-21](https://user-images.githubusercontent.com/20165848/94806462-a6310280-03f6-11eb-9ad6-b893e7584330.gif)

### Snippet

```rust
use std::io;

fn main() -> Result<(), io::Error> {
    write!(io::stdout().lock(), "input/ a folder with side_files");

    Ok(())
}
```

### Comments

After automatic restart of rust-analyzer it works without this issue but after manual restart it causes this in any way. 

I am not sure if it is a redundant issue.

### Version

`rust-analyzer version: 2020-09-28 (277488b)`

**Comment by bjorn3** [comment]
I also noticed this once today. I couldn't reproduce it though.

**Comment by matklad** [comment]
I can reproduce this on two-releases old rust-analyzer, but can't reproduce on master. Looks like fixed by #6222
